sql/functions.sql

```diff
@@ -236,6 +256,99 @@ END;
 $$
 LANGUAGE plpgsql IMMUTABLE;
 
+
+CREATE OR REPLACE FUNCTION get_postcode_rank(country_code VARCHAR(2), postcode TEXT,
+                                      OUT rank_search SMALLINT, OUT rank_address SMALLINT)
+AS $$
+DECLARE
+  part TEXT;
+BEGIN
+    rank_search := 30;
+    rank_address := 30;
+    postcode := upper(postcode);
+
+    IF country_code = 'gb' THEN
```

Comment:
This function basically determines at which administration level is covered by a single postcode. This is used when guessing how far away a postcode centroid can be and still be valid and when assigning a place to a postcode (i.e when you search by postcode, the result has a town or similar).

This is actually legacy code which I only have moved out of the placex_insert function. I'm not entirely sure if it is worth extending it for other countries. It needs to be reviewed at some point. There is one thing I forgot to put on the TODO list and that is recognizing postcodes by their pattern instead of taking it from the word list. This will be something where people can (and should) contribute country-specific knowledge.